modules/codegen/src/main/scala/com/twilio/guardrail/generators/Http4sServerGenerator.scala

```diff
@@ -85,19 +93,36 @@ object Http4sServerGenerator {
       case GenerateSupportDefinitions(tracing, securitySchemes) =>
         Target.pure(List.empty)
 
-      case RenderClass(resourceName, handlerName, _, combinedRouteTerms, extraRouteParams, responseDefinitions, supportDefinitions) =>
+      case RenderClass(resourceName, handlerName, _, routeTerms, authedRouteTerms, extraRouteParams, responseDefinitions, supportDefinitions) =>
         Target.log.function("renderClass")(for {
           _ <- Target.log.debug(s"Args: ${resourceName}, ${handlerName}, <combinedRouteTerms>, ${extraRouteParams}")
-          routesParams = List(param"handler: ${Type.Name(handlerName)}[F]")
-        } yield q"""
-          class ${Type.Name(resourceName)}[F[_]](..$extraRouteParams)(implicit F: Async[F]) extends Http4sDsl[F] {
-
+          types        = if (authedRouteTerms.isEmpty) List(Type.Name("F")) else List(Type.Name("F"), Type.Name("U"))
+          routesParams = List(param"handler: ${Type.Name(handlerName)}[..$types]")
+        } yield {
+          val authedRoutes =
+            if (authedRouteTerms.isEmpty) None
+            else Some(q"""
+            def authedRoutes[U](..${routesParams}): AuthedRoutes[U, F] = AuthedRoutes.of {
```

Comment:
Instead of defining a new `authedRoutes` function and requiring folk to call it, why not alter the return type of `routes` to yield a tuple of the non-authed routes as well as the authed routes?

If we take this approach, when a user switches on authed functionality, the compiler error makes it very obvious what needs to happen next.

Reply:
👍 Should be easy to change.